### ISSUE TITLE
Better support for prop function

### ISSUE THREAD

**Issue by apertureless** [simplesmiler/vue-clickaway#25]
I think a more general support for prop functions should be implemented.

Right now thats not working

```js
<template>
  <div class="popup-menu"
    v-if="open"
    v-on-clickaway="closeFn"
  >
    <div class="popup-menu__header">
      <button class="popup-menu__close" @click="closeFn()">
        Close
      </button>
    </div>
  </div>
</template>

<script>
  import { directive as onClickaway } from 'vue-clickaway'
  export default {
    directives: { onClickaway },
    props: {
       closeFn: {
          type: Function,
          default: function() {
             return this.close()
          }
       },
      methods: {
         close () {
            this.$emit('close')
         }
      }
    }
</script>
```

You have to wrap the `closeFn` into an anonymous function to work.

```js
v-on-clickaway="() => closeFn()"
```

However I find it pretty confusing because normal methods work without the anonymous function.

**Comment by simplesmiler** [comment]
Fixed with 2.2.2 release.

**Comment by acidjazz** [comment]
@apertureless and @simplesmiler should this fix also allow us to specify functions with parameters?

if I try to do something like

```jade
.dropdown(
  :class="{'is-active': controls.category}",
  @click="controls.category = !controls.category",
  v-on-clickaway="close('category')")
```

```js
close (control) {
  console.log(`close(${control})`)
  this.controls[control] = false
},
```

I get the error
```
[Vue warn]: v-on-clickaway="close('category')" expects a function value, got undefined
```

I have to write a `closeCategory()` function and pass it as `closeCategory`
